### ISSUE TITLE
[BUG] Password Length Limit

### ISSUE THREAD

**Issue by BJPickles** [suriyun-production/mmorpg-kit-docs#1715]
We've discovered an issue where the server will "timeout" if the password is too long.
We also found this was an issue with email (using the Firebase integration) however a potential fix seemed to be increasing the DB character limit for both username and password.

Even with the new limit, it still occasionally gives timeout errors.

![image](https://user-images.githubusercontent.com/48354382/193073536-1e724a47-12c9-448d-8c31-b47561162409.png)

**Comment by insthync** [comment]
How is the password encrypted?

Doesn't while it will be encrypted its length will be fixed?

It still has timeout errors, so it may have other error.

You should include log files when the issue occurs.

**Comment by shubhank008** [comment]
> How is the password encrypted?
> 
> Doesn't while it will be encrypted its length will be fixed?
> 
> It still has timeout errors, so it may have other error.
> 
> You should include log files when the issue occurs.

Password is encrypted using the kit's own internal system (md5 hash), we did not modify anything related to that.

I was trying to help @BJPickles through screenshare so wanted to pitch a few things I noticed.

1. Even if you change username limit in CentralNetworkManager to say something like 255, isn't it meaningless because the mysql database's username field is set to something like VARCHAR(32) or VARCHAR(40) ?  
As shown in above image, we had to manually modify the schema to change username to VARCHAR(255)

2. When @BJPickles tested, these kind of long passwords worked `earwaxanalogpushuprouses94`  / `123456789` / `Sblah@blah` but this kept throwing timeout error `EarwaxAnalogPushupRouses94`  

3. This should not be related to password length in MySQL as we changed its schema to VARCHAR(255) as well so unless the encrypted password is longer than that for `EarwaxAnalogPushupRouses94` then don't know the reason

4. There is nothing logged on client or server side, only the Timeout error which @BJPickles can provide you or screenshot.
Here is a video for the same - https://imgur.com/TxgjYSo

**Comment by insthync** [comment]
No logs files?

**Comment by insthync** [comment]
@BJPickles Did you try set more timeout duration?

**Comment by insthync** [comment]
As I look into the codes, it does not store anything in database, am I correct?

**Comment by insthync** [comment]
????

**Comment by BJPickles** [comment]
> @BJPickles Did you try set more timeout duration?

Apologies, I'm currently without Internet so just waiting for that to be fixed before I can test.

**Comment by insthync** [comment]
@BJPickles I can't find where it will make changes to the username, so I don't think username length is a cause of the problem.

Not sure that are you using this https://github.com/GamerzDan/MMORPGKIT-Firebase-AuthBackend ?

**Comment by BJPickles** [comment]
Yeah definitely using that. @shubhank008 (I believe the author) was helping me with it.

The only reason we noticed this was because I was creating an account for Steam to verify the game (and so they need an account to login ).

I attempted to create an account called "steamtester@mystudioemail.co.uk" and it failed to register, HOWEVER it bugged saying that it was registered somehow in the DB, despite not having any records.

We played around and found that a mixture of password and "username" (email) limit was rhe problem.

We could register short usernames and passwords, but after a certain length - if either the username or password was too long it would cause the same issue where the timeout would occur, and then it would fail to register the account, BUT somehow still have a record and not let the player re-register claiming the account had already been created.

We noticed that this occurred less after increasing the VAR to 255, but still occurred.

Beyond that I'm not sure why.
